### PR TITLE
Make circleci to run in parallel

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,10 +65,6 @@ install-tools:
 	  github.com/pavius/impi/cmd/impi \
 	  github.com/tcnksm/ghr
 
-.PHONY: otelcontribcol
-otelcontribcol:
-	GO111MODULE=on CGO_ENABLED=0 go build -o ./bin/otelcontribcol_$(GOOS)_$(GOARCH) $(BUILD_INFO) ./cmd/otelcontribcol
```
Comment (dmitryax) [inline Makefile:75]: why no `otelcontribcol` anymore? we have it in core as `otelcol`, looks like a reasonable target

**Review by dmitryax — APPROVED** [review]
Looks like circleci config replicated from core, right? Overall looks good. Just one nit.

File: Makefile (hunk 1 above)
Comment (bogdandrutu) [reply Makefile:75]: It is the same as binaries. Exactly same mistake in core. Why do we need 2 targets that do the same?

Comment (bogdandrutu) [reply Makefile:75]: binaries is nicer because it is the same between the repos ;)

**Comment by bogdandrutu** [comment]
Good enough, from ~12 mins to ~6 :)

File: Makefile (hunk 1 above)
Comment (dmitryax) [reply Makefile:75]: I usually run `make otelcol` to get `otelcol` binary :) 
`make build` would be a good option as well, what do you think?
And we still have `make docker-otelcol` :)

Comment (bogdandrutu) [reply Makefile:75]: Happy to review a PR to use otelcolcontrib and remove binaries, probably same in code

Comment (bogdandrutu) [reply Makefile:75]: Make build is better I think

**Pull request merged**